samcli/lib/build/utils.py

```diff
@@ -6,10 +6,49 @@
 
 from samcli.commands.local.lib.exceptions import OverridesNotWellDefinedError
 from samcli.lib.providers.provider import Function, LayerVersion
+from samcli.lib.build.build_graph import LayerBuildDefinition
+from samcli.lib.utils.architecture import X86_64, ARM64
 
 LOG = logging.getLogger(__name__)
 
 
+def valid_architecture(architecture: str) -> bool:
+    return architecture in [X86_64, ARM64]
+
+
+def warn_on_invalid_architecture(layer_definition: LayerBuildDefinition) -> None:
```

Comment:
Let's add some unit testing for this function. You can mock out the `LOG` object and assert that the `warn` method has the correct calls.